pkg/watch/mux.go

```diff
@@ -162,6 +165,7 @@ func (m *Broadcaster) loop() {
 		}
 		m.distribute(event)
 	}
+	m.distributing.Done()
```

Comment:
You are right, the channel sends are synchronous and cannot happen at this point. And they must all have been received by this point too, so even if closing a channel cancels blocked unbuffered channel sends - which I'm not sure about - this would never happen.

Nice catch!